lib/error/error_type_map.cpp

```diff
@@ -0,0 +1,73 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Pionix GmbH and Contributors to EVerest
+
+#include <utils/error/error_type_map.hpp>
+
+#include <utils/error.hpp>
+#include <utils/error/error_exceptions.hpp>
+#include <utils/yaml_loader.hpp>
+
+#include <everest/logging.hpp>
+
+namespace Everest {
+namespace error {
+
+ErrorTypeMap::ErrorTypeMap(std::filesystem::path error_types_dir) {
+    load_error_types(error_types_dir);
+}
+
+void ErrorTypeMap::load_error_types(std::filesystem::path error_types_dir) {
+    BOOST_LOG_FUNCTION();
+
+    if (!std::filesystem::is_directory(error_types_dir) || !std::filesystem::exists(error_types_dir)) {
+        throw EverestDirectoryNotFoundError(
+            fmt::format("Error types directory '{}' does not exist.", error_types_dir.string()));
+    }
+    for (const auto& entry : std::filesystem::directory_iterator(error_types_dir)) {
+        if (!entry.is_regular_file()) {
+            continue;
+        }
+        if (entry.path().extension() != ".yaml") {
+            continue;
+        }
+        std::string prefix = entry.path().stem().string();
+        json error_type_file = Everest::load_yaml(entry.path());
+        if (!error_type_file.contains("errors")) {
+            EVLOG_warning << "Error type file '" << entry.path().string() << "' does not contain 'errors' key.";
+            continue;
+        }
+        if (!error_type_file["errors"].is_array()) {
```

Comment:
```suggestion
        if (!error_type_file.at("errors").is_array()) {
```

Reply:
Is checked before, but doesn't hurt to use `.at` here